### PR TITLE
Make color picker available via the app menu

### LINKED ISSUE #2568: Make multi-element color picker available via the app menu
__What should we do?__

* [ ] Support color picker actions via the app menu (`Edit > Color`)

As a potential improvement, investigate if [adding a color icon](https://github.com/camunda/camunda-modeler-design/issues/52#issuecomment-973005198) is possible and an improvement

---

Proposed menu addition:

```
App menu > Edit > Color 
                        > White
                        > Blue
                        > Orange
                        > Green
                        > Red
                        > Purple
```

__Why should we do it?__

Support tool bar removal and general UI unification (https://github.com/camunda/camunda-modeler-design/issues/52).

### PR DESCRIPTION
**Pull request by smbea**
Closes #2568

__Screenshot:__

image

**Note:** electron only supports png or jpg files as menu item icons, hence why I add to a file for each color.

### CODE REVIEW

**Comment by smbea** [comment]
@andreasgeier let me know if you have any issue with the icons sizes

**Comment by nikku** [comment]
Also, let us rename `White` to `Default`, as this basically resets all set colors.

**Comment by smbea** [comment]
> @smbea Suggestion regarding the icons: Don't we want to add the respective border color, too?
> 
> That closely represents the actual look being set on selection and may provide higher contrast against the menu background:
> 
> image

Good point. I think it makes sense, specially when considering the white/default which has a different border color

**Comment by nikku** [comment]
Last request: Please also include the blueprint (SVG) that you used to generate the icons from. That helps us next time we consider to change them (look and feel wise).

**Comment by smbea** [comment]
Here is an updated screenshot with the new icons:



I also added the base svg icon file

**Comment by andreasgeier** [comment]
Looks good - I like it!

I assume that it also works in a multi-select state, right?

Please do not forget to check Windows and Linux as well.

**Comment by smbea** [comment]
> I assume that it also works in a multi-select state, right?

Yes, it works with various elements selected - behaviour remains the same.

> Please do not forget to check Windows and Linux as well.

- [x] Mac
- [x] Linux (@MaxTru)
- [ ] Windows (@philippfromme )

File: client/src/app/tabs/getEditMenu.js [hunk 1]
```diff
@@ -50,6 +77,28 @@ export function getAlignDistributeEntries({
...
+    submenu: COLORS.map(color => {
+      return {
+        label: `${color.title}`,
+        enabled: setColor,
+        action: 'setColor',
+        options: {
+          fill: color.fill,
+          stroke: color.stroke
+        },
+        icon: `app/resources/icons/${color.title.toLowerCase()}-circle.png`
```
Comment (nikku) [inline client/src/app/tabs/getEditMenu.js:96]: Proposal: Do not encode full icon paths here (in the client); rather just pass the name (with extension) and make the menu infrastructure retrieve it, relative to a search directory the app knows (i.e. resources).

I.e. in this case, `resources/icons/${...}-circle.png` could be enough.

Looking into the future, plug-ins could ship their own icons, too, via the `app-plugins` protocoll, i.e. `app-plugins://token-simulation/resources/some-icon.png`.

Cf. https://github.com/camunda/camunda-modeler/blob/develop/app/lib/plugins.js#L150.

**Review by nikku — COMMENTED** [review]
Just a minor thing to consider regarding icon (asset) paths.

**Comment by philippfromme** [comment]
electron_mjZJUA3DGM

@smbea Looks good on Windows. 👍🏻

File: client/src/app/tabs/getEditMenu.js (hunk 1 above)
Comment (smbea) [reply client/src/app/tabs/getEditMenu.js]: I updated the client to pass on `resources/icons/${...}-circle.png` and the menu builder to add the `app/` prefix to the path. Is this what you meant?

Comment (nikku) [reply client/src/app/tabs/getEditMenu.js]: Exactly. Not encoding the base path allows us to support multiple search paths in the future.

File: client/src/app/tabs/getEditMenu.js [hunk 2]
```diff
@@ -12,6 +12,33 @@ import Flags, { DISABLE_ADJUST_ORIGIN } from '../../util/Flags';
 
 const SPACE_KEY = ' ';
 
+const COLORS = [
+  {
+    title: 'Default',
+    fill: 'white',
```
Comment (nikku) [inline client/src/app/tabs/getEditMenu.js:18]: Proposal: Set `fill` and `stroke` to undefined instead to reset colors (to default, cf. https://github.com/bpmn-io/bpmn-js-color-picker/blob/master/colors/ColorPopupProvider.js#L20.

Comment (smbea) [reply client/src/app/tabs/getEditMenu.js]: Updated. Should be good to go now

**Review by nikku — APPROVED** [review]
Great stuff! :clap:

**Pull request merged**